student/src/main/resources/root-context.xml

```diff
@@ -153,14 +153,14 @@
 		</beans:property>
 
 		<beans:property name="poolName" value="AIRHikariCP" />
-		<beans:property name="maxLifetime" value="1800000" />
+		<beans:property name="maxLifetime" value="${datasource.maxLifetime:1800000}" />
 		<beans:property name="minimumIdle" value="${datasource.minPoolSize:5}" />
 		<beans:property name="maximumPoolSize" value="${datasource.maxPoolSize:20}" />
-		<!-- <beans:property name="leakDetectionThreshold" value="15000" /> -->
+		<beans:property name="connectionTimeout" value="${datasource.connectionTimeout:30000}" />
+		<beans:property name="leakDetectionThreshold" value="${datasource.leakDetectionThreshold:0}" />

```

Comment:
I made the leak detection, connection timeout, and maxLifetime settings configurable.